lib/pengine/complex.c

```diff
@@ -848,7 +847,7 @@ pe__unpack_resource(xmlNode *xml_obj, pe_resource_t **rsc,
                  (*rsc)->next_role != RSC_ROLE_UNKNOWN ? role2text((*rsc)->next_role) : "default");
 
     if ((*rsc)->fns->unpack(*rsc, data_set) == FALSE) {
-        return FALSE;
+        return pcmk_rc_unpack_error;
```

Comment:
The causes of failure vary, so we can't put them into one neat box.
```
Native:
* Resource of given standard can't be used as promotable clone

Group:
* Always returns TRUE

Clone:
* Empty child object
* Failure to create child

Bundle:
* No container child
* Invalid network
* Call to pe__unpack_resource() fails
* Container creation failure
```

Reply:
Yep, once we can change unpack() to return a standard code we can propagate that here. Unfortunately that's public API so it will have to wait a while.

Reply:
:'(